crates/task-impls/src/consensus/helpers.rs

```diff
@@ -596,14 +651,29 @@ pub(crate) async fn handle_quorum_proposal_recv<TYPES: NodeType, I: NodeImplemen
         debug!("Failed to update view; error = {e:#}");
     }
 
+    let mut parent_leaf = task_state
+        .consensus
+        .read()
+        .await
+        .saved_leaves()
+        .get(&justify_qc.date().leaf_commit)
+        .cloned();
+
+    parent_leaf = match parent_leaf {
+        Some(p) => Some(p),
+        None => fetch_proposal(
```

Comment:
I think this is fine, just wanted to check my understanding here: I guess we can block consensus here for up to `REQUEST_TIMEOUT` if the parent leaf is missing? But it shouldn't be a big deal because post-refactor we will have a separate task loop spawned per view, right?

Reply:
Yeah in the new version we can just block "forever" because the task will not block the main task